sys/net/gnrc/network_layer/ipv6/nib/nib.c

```diff
@@ -108,8 +108,10 @@ static inline bool _should_search_rtr(const gnrc_netif_t *netif)
     /* 6LBR interface does not send RS.
        A non-advertising router sends RS or a 6LN that is advertising or not
        has to refetch router information */
-    return !gnrc_netif_is_6lbr(netif) &&
-           (!gnrc_netif_is_rtr_adv(netif) || gnrc_netif_is_6ln(netif));
+    if (gnrc_netif_is_6lbr(netif)) {
+        return false;
+    }
+    return !gnrc_netif_is_rtr_adv(netif) || gnrc_netif_is_6ln(netif);
```

Comment:
I would prefer parens around the return statement - but that's probably personal taste.

Reply:
I like avoiding superfluous parens, what value do they add?

Reply:
IMO readability: the entire expression is evaluated first and then returned. Even so this is defined behavior I'm always in favor of making this explicit. I.e., rather one parenthesis too much than missing one. In general that helps to make sure that the precedence of the code is the one intended by the author.

Reply:
But there is no other way to evaluate this. Nothing can happen after the `return`, so there is no ambiguity. I agree that in large `if` statements it can help when operator precedence is not so obvious, but then again those should be avoided in general and rather broken off into multiple `if` statements to reduce the mental complexity for the reader.

But when there is no need, I'd rather avoid them to reduce the visual clutter.

Reply:
As said: personal taste. I think most important is to stay consistent. Within this file there are two other return statements with an expression that consists of more than one component: one does without and the other with parens. 🤷